aya/src/maps/xdp/dev_map.rs

```diff
@@ -0,0 +1,145 @@
+//! An array of network devices.
+
+use std::os::fd::AsRawFd;
+
+use aya_obj::generated::{bpf_devmap_val, bpf_devmap_val__bindgen_ty_1};
+
+use crate::{
+    maps::{check_bounds, check_kv_size, IterableMap, MapData, MapError},
+    sys::{bpf_map_lookup_elem, bpf_map_update_elem},
+    Pod,
+};
+
+/// An array of network devices.
+///
+/// XDP programs can use this map to redirect to other network
+/// devices.
+///
+/// # Minimum kernel version
+///
+/// The minimum kernel version required to use this feature is 4.14.
+///
+/// # Examples
+/// ```no_run
+/// # let mut bpf = aya::Bpf::load(&[])?;
+/// use aya::maps::xdp::DevMap;
+///
+/// let mut devmap = DevMap::try_from(bpf.map_mut("IFACES").unwrap())?;
+/// let source = 32u32;
+/// let dest = 42u32;
+/// devmap.set(source, dest, None::<i32>, 0);
+///
+/// # Ok::<(), aya::BpfError>(())
+/// ```
+#[doc(alias = "BPF_MAP_TYPE_DEVMAP")]
+pub struct DevMap<T> {
+    inner: T,
+}
+
+impl<T: AsRef<MapData>> DevMap<T> {
+    pub(crate) fn new(map: T) -> Result<DevMap<T>, MapError> {
+        let data = map.as_ref();
+        check_kv_size::<u32, bpf_devmap_val>(data)?;
+
+        let _fd = data.fd_or_err()?;
+
+        Ok(DevMap { inner: map })
+    }
+
+    /// Returns the number of elements in the array.
+    ///
+    /// This corresponds to the value of `bpf_map_def::max_entries` on the eBPF side.
+    pub fn len(&self) -> u32 {
+        self.inner.as_ref().obj.max_entries()
+    }
+
+    /// Returns the value stored at the given index.
+    ///
+    /// # Errors
+    ///
+    /// Returns [`MapError::OutOfBounds`] if `index` is out of bounds, [`MapError::SyscallError`]
+    /// if `bpf_map_lookup_elem` fails.
+    pub fn get(&self, index: u32, flags: u64) -> Result<DevMapValue, MapError> {
+        let data = self.inner.as_ref();
+        check_bounds(data, index)?;
+        let fd = data.fd_or_err()?;
+
+        let value = bpf_map_lookup_elem(fd, &index, flags).map_err(|(_, io_error)| {
+            MapError::SyscallError {
+                call: "bpf_map_lookup_elem".to_owned(),
+                io_error,
+            }
+        })?;
+        let value: bpf_devmap_val = value.ok_or(MapError::KeyNotFound)?;
+        Ok(value.into())
+    }
+
+    /// An iterator over the elements of the array. The iterator item type is `Result<u32,
+    /// MapError>`.
+    pub fn iter(&self) -> impl Iterator<Item = Result<DevMapValue, MapError>> + '_ {
+        (0..self.len()).map(move |i| self.get(i, 0))
+    }
+}
+
+impl<T: AsMut<MapData>> DevMap<T> {
+    /// Sets the value of the element at the given index.
+    ///
+    /// # Errors
+    ///
+    /// Returns [`MapError::OutOfBounds`] if `index` is out of bounds, [`MapError::SyscallError`]
+    /// if `bpf_map_update_elem` fails.
+    pub fn set(
```

Comment:
Same for all the other maps in the PR: the docs need better names (no generic
`value`) and need to explain how the `program` argument works

Reply:
Done in 3ef7c8b